base/iterators.jl

```diff
@@ -1343,4 +1344,144 @@ only(x::NamedTuple) = throw(
     ArgumentError("NamedTuple contains $(length(x)) elements, must contain exactly 1 element")
 )
 
+
+"""
+    logrange(start => stop, length)
+
+Returns an iterator which runs from `start` to `stop` with `length` elements
+spaced logarithmically, rather than linearly as for [`range`](@ref).
+That is, the ratio of successive elements is constant, not the difference.
+
+!!! compat "Julia 1.7"
+    This function requires at least Julia 1.7.
+
+This is similar to `geomspace` in Python, and to `PowerRange` in Mathematica.
+
+# Examples
+```jldoctest
+julia> foreach(println, logrange(2 => 16, 4))
+2.0
+4.0
+8.0
+16.0
+
+julia> collect(logrange(1000 => 1, 7))
+7-element Vector{Float64}:
+ 1000.0
+  316.22776601683796
+  100.0
+   31.622776601683793
+   10.0
+    3.1622776601683795
+    1.0
+
+julia> ans ≈ 10 .^ (3:-0.5:0)
+true
+
+julia> collect(logrange(-1 => -2f0, 3))
+3-element Vector{Float32}:
+ -1.0
+ -1.4142135
+ -2.0
+
+julia> map(rad2deg∘angle, logrange(1 => -1+0im, 5))
+5-element Vector{Float64}:
+   0.0
+  45.0
+  90.0
+ 135.0
+ 180.0
+```
+"""
+function logrange(fromto::Pair{<:Number, <:Number}, len::Integer)
+    len >= 2 || throw(ArgumentError("logrange must have length of at least 2"))
+    start, stop = map(float, promote(fromto...))
+    ratio = ratio_nth_root(stop, start, Int(len)-1)
+    LogRange(start, ratio, Int(len))
+end
```

Comment:
```suggestion
function logrange(start::Number, stop::Number; length::Integer)
    length>= 2 || throw(ArgumentError("logrange must have length of at least 2"))
    start, stop = map(float, promote(start, stop))
    ratio = ratio_nth_root(stop, start, Int(length)-1)
    LogRange(start, ratio, Int(length))
end
#38041 syntax, all keywords
logrange(; start::Number, stop::Number, length::Integer) = logrange(start, stop; length) 
#39069 syntax, pair syntax
logrange(fromto::Pair{<:Number, <:Number}, length::Integer) = logrange(fromto.first, fromto.second; length) 
```

Support standard and #38041 syntax